src/minecraftTransform.js

```diff
@@ -0,0 +1,66 @@
+import { Transform } from 'stream'
+import MapColor from './mapColors'
+
+export class ComputeColors extends Transform {
+	constructor() {
+		super()
+	}
+
+	// buffer to id array
+	_transform(chunk, _, next) {
+		MapColor.fromImage(chunk)
+			.then(this.push)
+			.then(next)
+	}
+}
+
+export class ComputeKnown extends Transform {
+	constructor(width, height) {
+		super({ readableObjectMode: true })
+		Object.assign(this, { width, height })
+		this.last = new Uint8Array(width * height)
+	}
+
+	_transform(chunk, _, next) {
+		let columns = {}
+		let rows = {}
+		for (let x = 0; x < this.width; x++)
+			for (let y = 0; y < this.height; y++) {
+				const index = x + this.width * y
+				if (chunk[index] !== this.last[index]) {
+					if (!columns.start) columns.start = x
+					if (!rows.start) rows.start = y
+					if (!rows.end || y > rows.end) rows.end = y
+					columns.end = x
+				}
```

Comment:
i shall see if there is a way to clean up this function a bit as i don't like the verbosity